File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;
stack<int> stk;

int solution(vector<vector<int>> board, vector<int> moves)
{
    int answer = 0;

    for (int i = 0; i < moves.size(); i++)
    {
        for (int j = 0; j < board[0].size(); j++)
        {
            if (board[j][moves[i] - 1] > 0 && board[j][moves[i] - 1] <= 100)
            {
                if (!(stk.empty()) && stk.top() == board[j][moves[i] - 1])
                {
                    stk.pop();
                    answer+=2;
                    board[j][moves[i] - 1] = 0;
                    break;
                }
                else
                {
                    stk.push(board[j][moves[i] - 1]);
                    board[j][moves[i] - 1] = 0;
                    break;
                }
            }
        }   
    }

    return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stk = new Stack<>();        
        
        for (var move : moves) {
            int col = move - 1;
            for (int row = 0; row < board[0].length; row++) {
                if (board[row][col] != 0) {
                    stk.push(board[row][col]);
                    board[row][col] = 0;
                    break;
                }
            }
            if (stk.size() >= 2 && stk.get(stk.size() - 1) == stk.get(stk.size() - 2)) {
                stk.pop(); stk.pop();
                answer += 2;
            }
        }
        
        return answer;
    }
}